core/sdk-core/src/main/java/software/amazon/awssdk/core/progress/listener/ProgressListener.java

```diff
@@ -0,0 +1,395 @@
+             * {@link GetObjectRequest}.
+             */
+            SdkHttpRequest request();
+
+            /**
+             * The immutable {@link ProgressSnapahot} for this specific update.
+             */
+            ProgressSnapshot progressSnapshot();
+        }
+
+        /**
+         * The submitted {@link SdkHttppRequest} request header was successfully sent to the service
+         * <p>
+         * Available context attributes:
+         * <ol>
+         *     <li>{@link RequestHeaderSent#request()}</li>
+         *     <li>{@link RequestHeaderSent#progressSnapshot()}</li>
+         * </ol>
+         */
+        @Immutable
+        @ThreadSafe
+        @SdkPublicApi
+        @SdkPreviewApi
+        public interface RequestHeaderSent extends RequestPrepared {
+        }
+
+        /**
+         * Additional bytes sent
+         * <p>
+         * Available context attributes:
+         * <ol>
+         *     <li>{@link RequestBytesSent#request()}</li>
+         *     <li>{@link RequestBytesSent#progressSnapshot()}</li>
+         * </ol>
+         */
+        @Immutable
+        @ThreadSafe
+        @SdkPublicApi
+        @SdkPreviewApi
+        public interface RequestBytesSent extends RequestHeaderSent {
+        }
+
+        /**
+         * Service has sent back a response header, denoting the start of response reception
+         * <p>
+         * Available context attributes:
+         * <ol>
+         *     <li>{@link ResponseHeaderReceived#request()}</li>
+         *     <li>{@link ResponseHeaderReceived#progressSnapshot()}</li>
+         * </ol>
+         */
+        @Immutable
+        @ThreadSafe
+        @SdkPublicApi
+        @SdkPreviewApi
+        public interface ResponseHeaderReceived extends RequestBytesSent {
+        }
+
+        /**
+         * Additional bytes received
+         * <p>
+         * Available context attributes:
+         * <ol>
+         *     <li>{@link ResponseBytesReceived#request()}</li>
+         *     <li>{@link ResponseBytesReceived#progressSnapshot()}</li>
+         * </ol>
+         */
+        @Immutable
+        @ThreadSafe
+        @SdkPublicApi
+        @SdkPreviewApi
+        public interface ResponseBytesReceived extends ResponseHeaderReceived {
+        }
+
+        /**
+         * The request execution is successful.
+         * <p>
+         * Available context attributes:
+         * <ol>
+         *     <li>{@link ExecutionSuccess#request()}</li>
+         *     <li>{@link ExecutionSuccess#progressSnapshot()}</li>
+         *     <li>{@link ExecutionSuccess#executionSuccess()} ()}</li>
+         * </ol>
+         */
+        @Immutable
+        @ThreadSafe
+        @SdkPublicApi
+        @SdkPreviewApi
+        public interface ExecutionSuccess extends ResponseBytesReceived {
+            /**
+             * The successful completion of a request submitted to the Sdk
+             */
+            ExecutionSuccessObjectRequest executionSuccess();
```

Comment:
Should we just "flatten" ExecutionSuccessObjectRequest into this type? E.g. add response() here and remove the ExecutionSuccessObjectRequest?

Reply:
Discussed offline, done.